mbed_build/_internal/find_files.py

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+from pathlib import Path
+from fnmatch import fnmatch
+from typing import Iterable
+
+
+def find_files(file_name: str, directory: Path) -> Iterable[Path]:
+    """Recursively find files matching name under given directory."""
+    return directory.rglob(file_name)
```

Comment:
Is it worth putting this in s function? To me if I see `Path.rglob(file)` it's easier to read than having to step down into this function to see what it's doing. Don't think it's really an issue either way but my preference would be to remove this function as I don't think it's really adding anything.

Reply:
Fair point, I'll do that.

Reply:
Actually, having thought about this - I think I'm going to leave it as is for now. 

Rationale being, the whole of file finding and exclusion is in a module that can guard the implementation details. I haven't done any performance testing (one of my worries). Having it this way means the return value of `find_files` might be swapped with a more clever object that `exclude` functions understand and are able to filter quickly.